application/src/main/java/org/thingsboard/server/service/cloud/KafkaUplinkMessageService.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright © 2016-2024 The Thingsboard Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.thingsboard.server.service.cloud;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.thingsboard.server.common.data.cloud.CloudEvent;
+import org.thingsboard.server.common.data.id.TenantId;
+import org.thingsboard.server.common.data.page.PageData;
+import org.thingsboard.server.dao.cloud.CloudEventService;
+
+@Slf4j
+public abstract class KafkaUplinkMessageService extends BaseUplinkMessageService {
+    @Autowired
+    protected CloudEventService cloudEventService;
+
+    public void processHandleMessages(TenantId tenantId) {
+        PageData<CloudEvent> cloudEvents = findCloudEvents(tenantId);
```

Comment:
max_poll_records: "${TB_QUEUE_KAFKA_MAX_POLL_RECORDS:8192}"

Based on the current config, you'll find 8192 records from Kafka and try to send them all at once. I think we could face rate limits (need to double-check it), but even without it, we need to split sending by some batch. Let's discuss it with @volodymyr-babak

Reply:
In my opinion and based on the tests I conducted, the best value is 200. It is the optimal value that yields good processing results. Of course, the fastest processing occurs with 8192.

Test results: 600,000 ts:

Value 200: processed in 4 minutes 24 seconds.
Value 8192: processed in 3 minutes 33 seconds.
During the test, 10,000 ts were sent every second for 60 seconds.